addon/components/select-2.js

```diff
@@ -463,6 +498,22 @@ var Select2Component = Ember.Component.extend({
         // treat data as a single object
         value = get(data, optionValuePath);
       }
+    } else if (this.get('tags') && e) {
+      if (e.added) {
+        var record;
+        if (e.added instanceof DS.Model) {
+          record = e.added;
+        } else {
+          record = this.createRecordFromTag(e.added);
+        }
+        this.get('value').pushObject(record);
+      } else if (e.removed) {
+        // TODO: Is this always the preferred behaviour? Maybe make this configurable...
+        this.get('value').removeObject(e.removed);
+        if (e.removed.get('isNew')) {
+          e.removed.unloadRecord();
+        }
```

Comment:
@iStefo What is your opinion on this? This way it will only unload records that are new in the store and persisted records are only removed from the collection. If this is not enough, I can make it configurable to always unload or only remove it from the collection and not from the store.